runtime/src/lib.rs

```diff
@@ -645,10 +645,19 @@ impl pallet_difficulty::Config for Runtime {
 	type Moment = Moment;
 }
 
+pub struct RewardIfNotPoS;
+
+impl pallet_rewards::RewardsEnabled for RewardIfNotPoS {
+	fn should_issue_rewards() -> bool {
+		pallet_pos_switch::SwitchBlockNumber::get().is_none()
```

Comment:
Causes error: 
`cannot infer type for type parameter `T` declared on the type alias `SwitchBlockNumber``

Reply:
The rest looks good to me.